File: google_interview.js
// Naive
function hasPairWithSum(arr, sum){
    var arrayElements = []
    var len = arr.length;
    for(var i =0; i<len-1; i++){
       for(var j = i+1;j<len; j++){
          if (arr[i] + arr[j] === sum)
          arrayElements.push(arr.indexOf(arr[i]))
          arrayElements.push(arr.indexOf(arr[j]))
          console.log(arrayElements)
          return true;
       }
    }
  
    return false;
  }
  
  // Better
  function hasPairWithSum2(arr, sum){
    const mySet = new Set();
    const len = arr.length;
    for (let i = 0; i < len; i++){
      if (mySet.has(arr[i])) {
        return true;
      }
      mySet.add(sum - arr[i]);
    }
    return false;
  }
  [1,2,4,4]
  hasPairWithSum2([6,4,3,2,1,7], 9)

File: google_interview.py
# Naive
def hasPairWithSum(arr:list , _sum: int) -> bool:
    for i in arr:
        for j in arr:
            if i + j == _sum:
                return [arr.index(i), arr.index(j)]
    return False


print(hasPairWithSum([1,2,3,7], 8))













# Better
# def hasPairWithSum2(arr, _sum):
#     myList = []
#     for i in arr:
#         if i in myList:
#             return True
#         myList.append(_sum - i)
#     return False
    

# print(hasPairWithSum2([1,2,4,4], 8))
